liquibase-core/src/main/java/liquibase/database/core/PostgresDatabase.java

```diff
@@ -296,4 +297,29 @@ private boolean runExistsQuery(String query) throws DatabaseException {
     public CatalogAndSchema.CatalogAndSchemaCase getSchemaAndCatalogCase() {
         return CatalogAndSchema.CatalogAndSchemaCase.LOWER_CASE;
     }
+
+    @Override
+    public String getDatabaseFullVersion() throws DatabaseException {
+        if (dbFullVersion != null){
+            return dbFullVersion;
+        }
+        final String sqlToGetVersion = "SELECT version()";
+        List<?> result = ExecutorService.getInstance().
+                getExecutor(this).queryForList(new RawSqlStatement(sqlToGetVersion), String.class);
+        if (result != null && !result.isEmpty()){
+            return result.get(0).toString();
+        }
+
+        return null;
+    }
```

Comment:
not sure how `getDatabaseFullVersion()` and `idEDB()` work together.
`getDatabaseFullVersion` never initializes **dbFullVersion** var, neither **idEDB** tries to call `getDatabaseFullVersion` if **dbFullVersion** is not initialized.

second, not sure you can take existence of `connection` for granted (can this method be called at any object life-cycle step?)

Reply:
1. getDatabaseFullVersion() and isEDB not work together. Those 2 methods just provide APIs

2. getDatabaseFullVersion() never initializes dbFullVersion var, neither idEDB tries to call getDatabaseFullVersion if dbFullVersion is not initialized. -- right. Because getDatabaseFullVersion () it's a method of Abstraction and isEDB() it's a specific method for PostgresDatabase. I'll refactor it a little bit just to get common sense

3. second, not sure you can take existence of connection for granted (can this method be called at any object life-cycle step?) -- As a robustness I'll add a check if we have a connection set at this point